### PR TITLE
Update DeepSpeed docs

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

I noticed that there was some holes in our documentation and examples, and after the default flags were changed (CPU Offloading turned off) we haven't updated the docs to reflect this. This fixes this + Clears up some of the ways to enable certain features. Also DeepSpeed had another release, meaning `mpi4py` isn't necessary to run.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/advanced/multi_gpu.rst [hunk 1]
```diff
@@ -697,24 +697,23 @@ To use DeepSpeed, you first need to install DeepSpeed using the commands below.
...
 
-ZeRO-Offload
-""""""""""""
+DeepSpeed ZeRO Stage 2
+""""""""""""""""""""""
 
-Below we show an example of running `ZeRO-Offload <https://www.deepspeed.ai/tutorials/zero-offload/>`_. ZeRO-Offload leverages the host CPU to offload optimizer memory/computation, reducing the overall memory consumption.
-For even more speed benefit, they offer an optimized CPU version of ADAM to run the offloaded computation, which is faster than the standard PyTorch implementation. By default we enable ZeRO-Offload.
+By default we enable DeepSpeed ZeRO Stage 2, which partitions your optimizer states (Stage 1) and your gradients (Stage 2) across your GPUs to reduce memory. In most cases this is more efficient or at parity with DDP, primarily due to the optimized custom communications written by the DeepSpeed team.
+As a result, benefits can also be seen on a single GPU. Do note that the default bucket sizes allocate around ``3.6GB`` of VRAM to use during distributed communications, which can be tweaked when instantiating the plugin described a few sections below.
```
Comment (rohitgr7) [inline docs/source/advanced/multi_gpu.rst:713]:
```suggestion
By default, we enable DeepSpeed ZeRO Stage 2, which partitions your optimizer states (Stage 1) and your gradients (Stage 2) across your GPUs to reduce memory. In most cases, this is more efficient or at parity with DDP, primarily due to the optimized custom communications written by the DeepSpeed team.
As a result, benefits can also be seen on a single GPU. Do note that the default bucket sizes allocate around ``3.6GB`` of VRAM to use during distributed communications, which can be tweaked when instantiating the plugin described in a few sections below.
```

File: docs/source/advanced/multi_gpu.rst [hunk 2]
```diff
@@ -752,11 +769,30 @@ You can also modify the ZeRO-Offload parameters via the plugin as below.
...
+    model = MyModel()
+    trainer = Trainer(gpus=4, plugins=DeepSpeedPlugin(cpu_offload=True), precision=16)
+    trainer.fit(model)
+
 
 Custom DeepSpeed Config
 """""""""""""""""""""""
 
-DeepSpeed allows use of custom DeepSpeed optimizers and schedulers defined within a config file. This allows you to enable optimizers such as `1-bit Adam <https://www.deepspeed.ai/tutorials/onebit-adam/>`_.
+In some cases you may want to define your own DeepSpeed Config, to access all parameters defined. We've exposed most of the important parameters, however there may be debug parameters to enable. Also, DeepSpeed allows use of custom DeepSpeed optimizers and schedulers defined within a config file which is supported.
```
Comment (rohitgr7) [inline docs/source/advanced/multi_gpu.rst:795]:
```suggestion
In some cases you may want to define your own DeepSpeed Config, to access all parameters defined. We've exposed most of the important parameters, however, there may be debugging parameters to enable. Also, DeepSpeed allows the use of custom DeepSpeed optimizers and schedulers defined within a config file that is supported.
```

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/advanced/multi_gpu.rst [hunk 3]
```diff
@@ -697,24 +697,23 @@ To use DeepSpeed, you first need to install DeepSpeed using the commands below.
...
     DeepSpeed currently only supports single optimizer, single scheduler within the training loop.
 
-ZeRO-Offload
-""""""""""""
+DeepSpeed ZeRO Stage 2
+""""""""""""""""""""""
 
-Below we show an example of running `ZeRO-Offload <https://www.deepspeed.ai/tutorials/zero-offload/>`_. ZeRO-Offload leverages the host CPU to offload optimizer memory/computation, reducing the overall memory consumption.
-For even more speed benefit, they offer an optimized CPU version of ADAM to run the offloaded computation, which is faster than the standard PyTorch implementation. By default we enable ZeRO-Offload.
+By default, we enable DeepSpeed ZeRO Stage 2, which partitions your optimizer states (Stage 1) and your gradients (Stage 2) across your GPUs to reduce memory. In most cases, this is more efficient or at parity with DDP, primarily due to the optimized custom communications written by the DeepSpeed team.
```
Comment (edenlightning) [inline docs/source/advanced/multi_gpu.rst:712]: add a reference link?

**Review by edenlightning — APPROVED** [review]

File: docs/source/advanced/multi_gpu.rst [hunk 4]
```diff
@@ -697,24 +697,23 @@ To use DeepSpeed, you first need to install DeepSpeed using the commands below.
...
-Additionally if you run into any issues installing m4py, ensure you have openmpi installed using ``sudo apt install libopenmpi-dev`` or ``brew install mpich`` before running ``pip install mpi4py``.
 
 .. note::
     Currently ``resume_from_checkpoint`` and manual optimization are not supported.
 
     DeepSpeed currently only supports single optimizer, single scheduler within the training loop.
 
-ZeRO-Offload
-""""""""""""
+DeepSpeed ZeRO Stage 2
```
Comment (Borda) [inline docs/source/advanced/multi_gpu.rst:709]: do we have somewhere also stage 1?

Comment (SeanNaren) [reply docs/source/advanced/multi_gpu.rst:709]: I mention in the first line this includes stage 1! pretty much all the time you'll use stage 2 which includes stage 1 :)

**Review by Borda — APPROVED** [review]
